superbuild/CMakeLists.txt

```diff
@@ -64,9 +82,42 @@ if (NOT PREBUILT_STAGE2 STREQUAL "NO" AND NOT PREBUILT_STAGE2_PATH STREQUAL "NO"
 endif()
 
 set(STAGE2_PATH "${CMAKE_BINARY_DIR}/stage2")
+if (NOT PREBUILT_STAGE1_PATH STREQUAL "NO")
+    set(STAGE2_PATH "${PREBUILT_STAGE1_PATH}")
+endif()
 if (NOT PREBUILT_STAGE2_PATH STREQUAL "NO")
     set(STAGE2_PATH "${PREBUILT_STAGE2_PATH}")
 endif()
+set(STAGE3_PATH "${CMAKE_BINARY_DIR}/stage3")
+
+if (NOT PREBUILT_STAGE1_PATH STREQUAL "NO" AND NOT PREBUILT_STAGE2_PATH STREQUAL "NO")
+    message(FATAL_ERROR "PREBUILT_STAGE1_PATH and PREBUILT_STAGE2_PATH is incompatible together.")
+endif()
+
+if (BUILD_STAGE1_TOOLCHAIN_ONLY OR BUILD_STAGE2_TOOLCHAIN_ONLY OR
+    BUILD_SPIRV_TRANSLATOR_ONLY OR BUILD_VC_INTRINSICS_ONLY OR BUILD_L0_LOADER_ONLY)
+    set(XE_DEPS OFF)
+endif()
+
+if (BUILD_SPIRV_TRANSLATOR_ONLY OR BUILD_VC_INTRINSICS_ONLY OR BUILD_L0_LOADER_ONLY)
+    set(INSTALL_WITH_XE_DEPS ON)
+endif()
+
+if (APPLE)
+    # XE are not supported on macOS targets
+    set(XE_DEPS OFF)
+
+    if (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86)|(X86)|(amd64)|(AMD64)")
+        set (MACOS_VERSION_MIN 10.12)
+    else ()
+        set (MACOS_VERSION_MIN 11.0)
+    endif()
+
+    if (APPLE AND ISPC_ANDROID_NDK_PATH STREQUAL "NO")
```

Comment:
`ISPC_ANDROID_NDK_PATH` is needed only when building Linux based targets (Linux, Android).

Reply:
Hm, that's really strange because I somehow concluded that NDK is needed on macOS when ISPC_CROSS is on.

Reply:
Cross triggers Android build.

Reply:
The thing is that with `-DISPC_CROSS=ON` and without `ISPC_ANDROID_NDK_PATH` on `APPLE`, cmake reports this error:
https://github.com/ispc/ispc/blob/d3f561cc742752acbb4ff0c0a517c615800c049b/CMakeLists.txt#L98

Reply:
Ok, so these are different default setting here and in ISPC Cmake, so here we just enabling CROSS and it triggers Android build, but here we are not managing that directly. It's confusing, but ok, let it be this way.